src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java

```diff
@@ -348,31 +341,19 @@ private ExtractExecutor.Arguments getValidatedArguments() {
     if (needJdbcSchemas != null) {
       argumentsBuilder.setNeedJdbcSchemas(needJdbcSchemas);
     }
-    // prevRunPath is set only when the specified mode is not NORMAL.
-    if (mode.equals(RunMode.INCREMENTAL)) {
-      if (chunkRows < 1) {
-        throw new ParameterException(
-            spec.commandLine(),
-            "Non-normal run modes require chunked processing. Set --rows-per-chunk to a positive"
-                + " integer to enable chunked processing.");
-      }
-      if (prevRunPathString == null) {
-        throw new ParameterException(
-            spec.commandLine(), "--run-mode is not NORMAL but --prev-run-path is unspecified.");
-      }
-      Path path = Paths.get(prevRunPathString);
-      if (path.toString().endsWith(".zip")) {
-        throw new ParameterException(
-            spec.commandLine(), "Incremental mode is not supported for zipped records, yet.");
-      }
-      if (!Files.isDirectory(path)) {
-        throw new ParameterException(
-            spec.commandLine(),
-            String.format(
-                "--prev-run-path must specify a directory, but '%s' is not a directory.", path));
-      }
-      argumentsBuilder.setPrevRunPath(path);
+    switch (mode) {
+      case INCREMENTAL:
+        validateAndSetPrevRunPathIncrementalMode();
```

Comment:
`getValidatedArguments` becomes a majority of useful code existing in this class. Also, most of the validation is repetitive (check ending, check if path or parent is a directory).

My suggestion is to move the validation logic to a separate class with a clear control flow (e.g. based on the parameter name), maybe in a separate CL.

Reply:
Makes sense. I also think the naming of this class itself is a bit confusing. Will do the housekeeping work in another CL.